src/mlang/test_framework/test_interpreter.ml

```diff
@@ -120,51 +120,6 @@ let to_MIR_function_and_inputs (program : Bir.program) (t : test_file)
   ( { func_variable_inputs; func_constant_inputs; func_outputs; func_conds },
     input_file )
 
-let add_test_conds_to_combined_program (p : Bir.program)
-    (conds : Bir.condition_data Bir.VariableMap.t) : Bir.program =
-  (* because evaluate_program redefines everything each time, we have to make
-     sure that the redefinitions of our constant inputs are removed from the
-     main list of statements *)
```

Comment:
I suggest we remove this function (only used in this file) as it looks like it has no more use. When it's called, the conditions context statement have already been added, and the context variables initialisation is done with SimpleVar type so no InputVar needs to be filtered and replaced.

Reply:
Ok!